fastaparse.py

```diff
@@ -0,0 +1,179 @@
+                result[seq_id] = ""
+            else:
+                result[seq_id] = f'{result[seq_id]}{line.strip()}'
+    return result
+
+
+def translate(rna_seq):
+    """
+    Function translates input RNA sequence to protein sequence
+    :param rna_seq: string which represents target RNA sequence (it can start from AUG (M) or not,
+    but it must have length multiplied by 3)
+    :return: protein sequence for input RNA sequence
+    """
+    rna_seq = re.sub(" +", '', rna_seq.strip())
+    assert not len(rna_seq) % 3, "Length of RNA sequences must multiple by 3"
+    assert not 'T' in rna_seq, "RNA sequences can't contain Tymin"
+    protein_seq = ""
+    for codon in range(0, len(rna_seq), 3):
+        protein_seq = f'{protein_seq}{codons[rna_seq[codon: codon + 3]]}'
+    return protein_seq
+
+
+def calc_mass(rna_seq):
+    """
+    Function receives RNA sequence as input and returns mass for encoded protein
+    :param rna_seq: string which represents target RNA sequence (it can start from AUG (M) or not,
+    but it must have length multiplied by 3)
+    :return: summary mass for protein encoded by input RNA
+    """
+    rna_seq = re.sub(" +", '', rna_seq.strip())
+    assert not len(rna_seq) % 3, "Length of RNA sequences must multiple by 3"
+    assert not 'T' in rna_seq, "RNA sequences can't contain Tymin"
+    mass = float(0)
+    for codon in range(0, len(rna_seq), 3):
+        aa = codons[rna_seq[codon: codon + 3]]
+        mass += mass_table[aa]
+    return mass
+
+
+def orf(rna_seq):
+    """
+    Function receives RNA sequence as input and returns all possible ORFs
+    If for one start codon there are several stops, the first one will be chosen
+    :param rna_seq: string which represents target RNA sequence (it can start from AUG (M) or not,
+    but it must have length multiplied by 3)
+    :return: all possible ORFs within input RNA sequence (if for one start codon there are several stops,
+    the first one will be chosen)
+    """
+    assert not len(rna_seq) % 3, "Length of RNA sequences must multiple by 3"
+    assert not 'T' in rna_seq, "RNA sequences can't contain Tymin"
+    rna_seq = re.sub(" +", '', rna_seq.strip())
+    starts = list()
+    stops = list()
+    seqs = list()
+
+    for codon in range(0, len(rna_seq) - 2):
+        if codons[rna_seq[codon: codon + 3]] == "M":
+            starts.append(codon)
+        elif codons[rna_seq[codon: codon + 3]] == "Stop":
+            stops.append(codon)
+    starts.sort()
+    stops.sort()
+    for start in starts:
+        for stop in stops:
+            if start < stop and not ((stop - start) % 3) and (stop - start > 3):
+                seq = list()
+                for codon in range(start, stop - 6, 3):
+                    if codons[rna_seq[codon: codon + 3]] == "Stop":
+                        break
+                    seq.append(codons[rna_seq[codon: codon + 3]])
+                seqs.append("".join(map(str, seq)))
```

Comment:
And if you will do like this you can also avoid this very nested and unreadable construction, because since you have a protein string, you can easily add it as list element:
```suggestion
                seqs.append(seq)
```